src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -15296,177 +15282,182 @@ namespace SurfaceGeometry {
                     }
                 }
             }
         }
 
-        M = 0;
-        SurfCollinearWarning = false;
-        for (n = 1; n <= NSides; ++n) { // perform convexity test in the plane determined above.
-            V1len = std::sqrt(pow_2(A(n + 1) - A(n)) + pow_2(B(n + 1) - B(n)));
-            V2len = std::sqrt(pow_2(A(n + 2) - A(n + 1)) + pow_2(B(n + 2) - B(n + 1)));
-            if (V1len <= 1.e-8 || V2len <= 1.e-8) continue;
-            DotProd = (A(n + 1) - A(n)) * (B(n + 2) - B(n + 1)) - (B(n + 1) - B(n)) * (A(n + 2) - A(n + 1));
-            cosarg = DotProd / (V1len * V2len);
+        for (int n = 1; n <= NSides; ++n) { // perform convexity test in the plane determined above.
+
+            DataVectorTypes::Vector_2d pt0(A(n), B(n));
+            DataVectorTypes::Vector_2d pt1(A(n + 1), B(n + 1));
+            DataVectorTypes::Vector_2d pt2(A(n + 2), B(n + 2));
+
+            DataVectorTypes::Vector_2d V1 = pt1 - pt0;
+            DataVectorTypes::Vector_2d V2 = pt2 - pt1;
+
+            Real64 V1len = V1.length(); // = norm_L2()
+            Real64 V2len = V2.length();
+            if (V1len <= 1.e-8 || V2len <= 1.e-8) {
+                // At least two points are coincident. Should this happen? GetVertices is supposed to pop these vertices
+                continue;
+            }
+            Real64 DotProd = V1.dot(V2);
+            Real64 cosarg = DotProd / (V1len * V2len);
             if (cosarg < -1.0) {
                 cosarg = -1.0;
             } else if (cosarg > 1.0) {
                 cosarg = 1.0;
             }
-            Theta = std::acos(cosarg);
-            if (Theta < (ACosZero - TurnThreshold)) {
+            Real64 Theta = std::acos(cosarg);
+            if (Theta > TurnThreshold) {
                 SignFlag = true;
-            } else {
-                if (Theta > (ACosZero + TurnThreshold)) {
-                    SignFlag = false;
-                } else { // Store the index of the collinear vertex for removal
-                    if (!SurfCollinearWarning) {
-                        if (state.dataGlobal->DisplayExtraWarnings) {
-                            ShowWarningError(state,
-                                             "CheckConvexity: Surface=\"" + state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Name +
-                                                 "\", Collinear points have been removed.");
-                        }
-                        SurfCollinearWarning = true;
-                    }
-                    ++state.dataErrTracking->TotalCoincidentVertices;
-                    ++M;
-                    SurfCollinearVerts(M) = n + 1;
-                    continue;
+            } else if (Theta < -TurnThreshold) {
+                SignFlag = false;
+            } else { // std::abs(Theta) < TurnThreshold
+                // Store the index of the collinear vertex for removal
```

Comment:
Previous so called `DotProd` was actually a cross product, which required using this ACosZero value. Convert to an actual dot product, and use Vector_2d for clarity (and avoid reimplementing the dot product manually once again)